### ISSUE TITLE
[4.0.0-beta.12] - - View sometimes doesn’t update on data change from subscribe

### ISSUE THREAD

**Issue by bkarv** [ionic-team/ionic#15855]
# Bug Report

**Ionic Info**
Run `ionic info` from a terminal/cmd prompt and paste the output below.

```
Ionic:

   ionic (Ionic CLI)          : 4.2.0 (/usr/local/lib/node_modules/ionic)
   Ionic Framework            : @ionic/angular 4.0.0-beta.12
   @angular-devkit/core       : 0.9.0-beta.3
   @angular-devkit/schematics : 0.9.0-beta.3
   @angular/cli               : 6.1.5
   @ionic/ng-toolkit          : 1.1.0
   @ionic/schematics-angular  : 1.0.7

Cordova:

   cordova (Cordova CLI) : 7.1.0
   Cordova Platforms     : none
   Cordova Plugins       : no whitelisted plugins (5 plugins total)

System:

   ios-deploy : 1.9.1
   NodeJS     : v10.10.0 (/usr/local/bin/node)
   npm        : 6.4.1
   OS         : macOS High Sierra
```

**Describe the Bug**
I am getting a very weird error in Ionic 4 where my view doesn’t update automatically when the data changes via a API subscribe response. However after the API has pulled the new data if I then click on the html view the data updates (almost like I am setting focus to the html view).

The issue with the error is very inconsistent and hard to replicate. But it now has happened twice to me when migrating my components.

A bit of background I have a component that displays the role hierarchy of an organisation. Below is what the data object looks like. It is a nested objects.

```
let obj1: OrganisationRoleChart = {
      id: "59f4d75d-5dd3-401c-a313-572efb266aec",
      organisation: {id: "e36d1d62-c7ab-498d-8fc9-c674b7eb6b23", name: "ABCCOMPANY", 
organisation_type: {id: 10, name: "SUPPLIER"}, default_folder: null},
      role: {id: 2, name: "CEO", description: "Head of finance", is_protected: false},
      description: null,

        children: [{id: "c1688286-2eeb-4c32-901e-f00788b31b5d",
           organisation: {id: "e36d1d62-c7ab-498d-8fc9-c674b7eb6b23", name: "ABCCOMPANY", 
           organisation_type: {id: 10, name: "SUPPLIER"}, default_folder: null},
           role: {id: 3, name: "Accountant", description: "Account dept", is_protected: false},
           description: null,

          children: []}]
    }
```

<img width="452" alt="screen shot 2018-10-04 at 8 19 41 pm" src="https://user-images.githubusercontent.com/10624315/46502930-b5fbc780-c86c-11e8-9754-8b0d4e034b05.png">

When I make a change like removing a role, it performs the action and the calls an API which is part of shared service to retrieve the updated role chart as below.

```
/*
  ** name: Set Record
  ** desc: 
  */
  setRecord(): void {
    this.isLoad = true;
    this.presentLoading('Fetching...');
    console.log('Fetching role chart...');
    this.roleResourceService.getOrganisationRolesChartsHttp(this.id)
    .subscribe(
      data => { 
        console.log('Role charts fetched!', data);
        if (data.length > 0) {
          this.record = data[0];
        } else {
          this.record = null;
        }      
        this.dismissLoading();
        this.isLoad = false;
      },
      err => {
        this.events.publish('alert:server-error', err);              
        this.dismissLoading();
        this.isLoad = false;
      },
      () => console.log('Request Complete')
    );
```
**The issue is when the new data is retrieved the view doesn’t update with the new hierarchy. But the weird thing is after pulling the new data when I click on the component area in the html view it then updates with the latest changes.**

I did some further testing and made a simple button which manually changes the record. This works and the view updates as expected.

```
test(): void {
    let obj1: OrganisationRoleChart = {
      id: "59f4d75d-5dd3-401c-a313-572efb266aec",
      organisation: {id: "e36d1d62-c7ab-498d-8fc9-c674b7eb6b23", name: "ABCCOMPANY", organisation_type: {id: 10, name: "SUPPLIER"}, default_folder: null},
      role: {id: 2, name: "CEO", description: "Head of finance", is_protected: false},
      description: null,
         children: [{id: "c1688286-2eeb-4c32-901e-f00788b31b5d",
          organisation: {id: "e36d1d62-c7ab-498d-8fc9-c674b7eb6b23", name: "ABCCOMPANY", 
          organisation_type: {id: 10, name: "SUPPLIER"}, default_folder: null},
          role: {id: 3, name: "Accountant", description: "Account dept", is_protected: false},
          description: null,
      children: []}]
    }
    this.record = obj1;
  }
```
So I am absolutely lost why is this happening, is as if the view component is loosing focus when it is does the API call. Has anyone else run into a similar issue? Appreciate any help on this.

**Steps to Reproduce**
This is the issue, the error is so inconsistent it is very hard to reproduce. All I can determine in the above scenario it seems to be happening when I use the API subsribe. 

Just to reiterate this code worked without any issues in Ionic 3 so there is some underlying bug in Ionic 4 causing this issue.

**Comment by bkarv** [comment]
To anyone having a similar issue where the the view doesn't update I was given the following solution from the forums which resolved the issue. That is I forced a manual check.

Imported change detector 
```
import { ChangeDetectorRef } from '@angular/core'

constructor(private changeRef: ChangeDetectorRef)
```

Then where I change the record where the view should updated I added the following code:

`this.changeRef.detectChanges();`

So in this instance it was my setRecord function:

```
setRecord(): void {
    this.isLoad = true;
    this.presentLoading('Fetching...');
    console.log('Fetching role chart...');
    this.roleResourceService.getOrganisationRolesChartsHttp(this.id)
    .subscribe(
      data => { 
        console.log('Role charts fetched!', data);
        if (data.length > 0) {
          this.record = data[0];
        } else {
          this.record = null;
        }      
        this.dismissLoading();
        this.isLoad = false;
        this.changeRef.detectChanges();
      },
      err => {
        this.events.publish('alert:server-error', err);              
        this.dismissLoading();
        this.isLoad = false;
      },
      () => console.log('Request Complete')
    );
  }
```

This is a solution but I still think there is underlying problem with Ionic 4/Angular where not all changes are automatically detected.

**Comment by simion** [comment]
Having the same issue.
I also implemented @bkarv before findind this issue, and it works ok.

**Comment by bkarv** [comment]
Seems to be fixed now in Ionic 4.0.0-beta.15. I am closing this.
